eclair-core/src/test/scala/fr/acinq/eclair/payment/MultiPartPaymentLifecycleSpec.scala

```diff
@@ -698,13 +699,13 @@ object MultiPartPaymentLifecycleSpec {
   val channelUpdate_ad = defaultChannelUpdate.copy(shortChannelId = channelId_ad)
   val channelUpdate_de = defaultChannelUpdate.copy(shortChannelId = channelId_de)
 
-  val hop_ab_1 = ChannelHop(a, b, channelUpdate_ab_1)
-  val hop_ab_2 = ChannelHop(a, b, channelUpdate_ab_2)
-  val hop_be = ChannelHop(b, e, channelUpdate_be)
-  val hop_ac_1 = ChannelHop(a, c, channelUpdate_ac_1)
-  val hop_ac_2 = ChannelHop(a, c, channelUpdate_ac_2)
-  val hop_ce = ChannelHop(c, e, channelUpdate_ce)
-  val hop_ad = ChannelHop(a, d, channelUpdate_ad)
-  val hop_de = ChannelHop(d, e, channelUpdate_de)
+  val hop_ab_1 = ChannelHop(channelUpdate_ab_1.shortChannelId, a, b, ChannelRelayParams.FromAnnouncement(channelUpdate_ab_1))
+  val hop_ab_2 = ChannelHop(channelUpdate_ab_2.shortChannelId, a, b, ChannelRelayParams.FromAnnouncement(channelUpdate_ab_2))
+  val hop_be = ChannelHop(channelUpdate_be.shortChannelId, b, e, ChannelRelayParams.FromAnnouncement(channelUpdate_be))
+  val hop_ac_1 = ChannelHop(channelUpdate_ac_1.shortChannelId, a, c, ChannelRelayParams.FromAnnouncement(channelUpdate_ac_1))
+  val hop_ac_2 = ChannelHop(channelUpdate_ac_2.shortChannelId, a, c, ChannelRelayParams.FromAnnouncement(channelUpdate_ac_2))
+  val hop_ce = ChannelHop(channelUpdate_ce.shortChannelId, c, e, ChannelRelayParams.FromAnnouncement(channelUpdate_ce))
+  val hop_ad = ChannelHop(channelUpdate_ad.shortChannelId, a, d, ChannelRelayParams.FromAnnouncement(channelUpdate_ad))
+  val hop_de = ChannelHop(channelUpdate_de.shortChannelId, d, e, ChannelRelayParams.FromAnnouncement(channelUpdate_de))
```

Comment:
You could use `channelHopFromUpdate`

Reply:
Nice catch, done in 623a71182. I'm not proud of `channelHopFromUpdate`, I want to get rid of it eventually but for now just cleaned up `Router.scala`.